### PR TITLE
add instructor outlines for OOP parts 1 & 2

### PR DESCRIPTION
**Pull request by pab2163**
Part 1: overview of procedural programming vs. OOP, creating classes
Part 2: inheritance

### CODE REVIEW

File: lessons/24_inheritence/inheritance_instructor_outline.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    def __init__(self, volume):
+        super().__init__(volume)
+
+    def play(self):
+        print('boom boom bap!')
+```
+
+Neither `Guitar` nor `Instrument` have the `play()` method yet
+
+# Wrap-up
```
Comment (ashiq-r31) [inline lessons/24_inheritence/inheritance_instructor_outline.md:186]: I think when discussing inheritance it also helps to highlight some complexities revolving multiple inheritances. 

This [article](https://medium.com/the-kickstarter/multiple-inheritance-in-python-84258756e04c) shows some of those complexities as well as shows why we might need `super()`

Comment (pab2163) [reply lessons/24_inheritence/inheritance_instructor_outline.md:186]:
```suggestion

# Multiple inheritance

* Demonstrate that one class can inherit from *multiple* other parent classes
* `Guitar` inherits from `Instrument` and `Luggage`

```python
class Luggage():
    item_type = 'luggage'
    def __init__(self, weight):
        self.weight = weight

class Guitar(Instrument, Luggage):
    def __init__(self, volume, electric, weight):
        Instrument.__init__(self, volume)
        Luggage.__init__(self, weight)
        self.electric = electric

    def display_info(self):
        print(f'Guitar set to volume {self.volume}')
        if self.electric:
            print('Guitar is electric!')
        else:
            print('Guitar is acoustic!')
            
g = Guitar(volume = 11, electric = False, weight = 20)

print(g.electric, g.volume, g.weight, g.item_type)
```


# Wrap-up
```

File: lessons/23_oop/oop_instructor_outline.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        return f'${cart_total}'
+    
+cart = Cart()
+cart.add('oreos', 12)
+cart.add('bananas', 2)
+
+print(cart.total())
+```
+
+# Final thoughts
```
[the remark this answers is not in the document]
Comment (pab2163) [reply lessons/23_oop/oop_instructor_outline.md:200]: Sure -- I've never used this before. Is the use case always when we want to hard code an attribute common to all instances of the class?

File: lessons/24_inheritence/inheritance_instructor_outline.md (hunk 1 above)
Comment (ashiq-r31) [reply lessons/24_inheritence/inheritance_instructor_outline.md]: I think the main thing I wanted to highlight was that multiple inheritance is **possible** (lifting this from another article):

```
class B:
    def b(self):
        print('b')


class C:
    def c(self):
        print('c')


class D(B, C):
    def d(self):
        print('d')


d = D()
d.b()
d.c()
d.d()
```
But **gets problematic** once parents have methods with the names leading to the MRO problem, link them to resources if they want to dig deeper. 
And finally, stress that for the rest of the program we will stick to single inheritance.

**Pull request merged**